collection.go

```diff
@@ -135,3 +170,52 @@ func (c *Client) GetUserCollections() (*[]Collection, error) {
 	}
 	return colls, nil
 }
+
+// DeleteCollection permanently deletes a collection and makes any posts on it
+// anonymous.
+//
+// See https://developers.write.as/docs/api/#delete-a-collection.
+func (c *Client) DeleteCollection(p *DeleteCollectionParams) error {
+	endpoint := "/collections/" + p.Alias
+	env, err := c.delete(endpoint, nil /* data */)
+	if err != nil {
+		return err
+	}
+
+	status := env.Code
+	switch status {
+	case http.StatusNoContent:
+		return nil
+	case http.StatusUnauthorized:
+		return fmt.Errorf("Not authenticated.")
+	case http.StatusBadRequest:
+		return fmt.Errorf("Bad request: %s", env.ErrorMessage)
+	default:
+		return fmt.Errorf("Problem deleting collection: %d. %s\n", status, env.ErrorMessage)
+	}
+}
+
+// CollectPosts adds a group of posts to a collection.
+//
+// See https://developers.write.as/docs/api/#move-a-post-to-a-collection.
+func (c *Client) CollectPosts(sp *CollectPostParams) ([]*CollectPostResult, error) {
+	endpoint := "/collections/" + sp.Alias + "/collect"
+
+	var p []*CollectPostResult
+	env, err := c.post(endpoint, sp.Posts, &p)
```

Comment:
no need to pass the address here as slices are reference types already

Reply:
You need to pass a pointer for the JSON unmarshaler to fill it with a decoded value. So json.Unmarshal requires the result argument to always be a pointer.

https://play.golang.org/p/_eT_ICsK6dt